Architecture_Engine/Modify/SetOutlineElements1D.cs

```diff
@@ -0,0 +1,52 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Architecture.Elements;
+using BH.oM.Dimensional;
+using BH.oM.Geometry;
+using System.Collections.Generic;
+using System.Linq;
+
+using BH.oM.Reflection.Attributes;
+using System.ComponentModel;
+
+namespace BH.Engine.Architecture
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /****               Public Methods              ****/
+        /***************************************************/
+
+        [Description("Assign a new collection of external 1D boundaries to an Architecture Room")]
+        [Input("room", "An Architecture Room to update")]
+        [Input("outlineElements1D", "A collection of outline 1D elements to assign to the Room")]
+        [Output("room", "The updated Architecture Room")]
+        public static Room SetOutlineElements1D(this Room room, List<IElement1D> outlineElements1D)
+        {
+            Room r = room.GetShallowClone() as Room;
+            r.Perimeter = outlineElements1D.Cast<ICurve>().ToList()[0];
+            return r;
+        }
```

Comment:
at least try to join the incoming IElement1ds and if the Room definition requires it, check for closed-ness